### DISCUSSION TITLE
8월 9일 시즌3 요구사항 도출 워크샵 1

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#217] [Notice]
## 할 일
- 기획 - 비전, 미션, 목적, 원칙을 통해 시즌 3의 목표를 만든다
- 기능 - 구현할 기능 목록을 뽑고 우선순위로 구현할 기능을 선발한다
- 스토리 - 기능이 사용되는 스토리(상황)를 정리한다
- 시나리오 - 스토리의 세부 시나리오를 스텝별(given, when, then)로 정리한다

### 공통
- 3인으로 조를 만든다
  - 엘라, 나봄, 현구막
  - 서니, 티케, 손너잘
  - 웨지, 디토, 조앤
- 각 단계 활동을 함께 논의한다

### 기획
- 조별로 비전, 미션, 목적, 원칙을 작성하여 취합한다.
- 취합한 의견을 위키로 정리한다.

### 기능 
- 기획을 달성하기 위해 필요한 기능을 취합한다.
- 기능들의 우선순위를 매겨서 구현할 기능을 선발한다.

### 스토리
- 기능이 필요한 상황을 취합한다.
- 역할별로 스토리가 달라질 수 있다.
- 상황을 우선순위에 매겨서 기록한다.

### 시나리오
- 스토리에서 세부 스텝을 정한다.

**Comment by sihyung92** [comment]
### 프로젝트에 참가한 이유

웨지 : 이미지 삽입 등 아직 구현 못한 기능들을 다듬고 싶다. 인프라도 만져보고 싶다.

디토 : 레벨3 바쁠줄 알고 나갔다가 그리워져서 돌아옴(철새) 보니까 버그가 몇개 있었다. 버그도 고치고 저번에 싸놓은 똥을 치우기 위해 돌아왔다. 

조앤 : 레벨4 때 면접이 있기 때문에, 프롤로그 프로젝트를 다듬어서 제대로 된 서비스를 제공하는 경험을 해보고 싶다. 지금 하고 있는 프로젝트도 비슷한 내용이라 해당 경험을 활용할 수 있을 것 같다

### 비전

베타에서 벗어나 정식 서비스로 발돋움하기

### 미션

- 이미지 삽입 구현과 유저 불편사항을 개선한다 + 무중단 배포
- 버그없이 CRUD가 작동할 수 있도록 이슈 해결하기 (마이페이지에서 내가 작성한 페이지가 다 안 나온다던지 하는 이슈들 해결)
- 학습로그를 시각화 한다

### 목적

> 비전, 미션과 관련있어야 하고, 측정이 가능하고 시간내에 성취할 수 있는 구체적인 내용으로 작성하기

- 버그없이 이미지 삽입이 가능하다.
- 무중단 배포 서비스가 된다.
- 이슈란에 버그성 이슈를 모두 해결한다.
- 시각화 기능을 추가한다 (방법 구체화 필요)

### 원칙

- 시각적 성취감을 부여해야 한다.
- 작성 편의성이 좋아야 한다.
- 숙제같은 느낌을 제거하여 사용자가 부담을 느끼지 않도록 해야 한다.
- 코드 품질 유지를 위해 기능 추가 이후 코드리뷰 시간을 가진다.
- 서로 의사소통이 원활하게 되어야 한다.

**Comment by bucketHaneul** [comment]
# 비전
- 엘라) 크루들이 본인의 TIL을 분산해서 작성을 하는데(노션, 깃헙, 블로그,,,), 이거를 우리 프롤로그에서 작성하게 한다. 프롤로그가 주요 작성지가 되게한다.
- 나봄) 나만을 위한 장이 아닌, 크루들 간의 정보를 공유하는 장이 되면 좋겠다.
- 나봄) 이것 자체가 포트폴리오가 되면 좋겠다.
- 현구막) 프롤로그를 작성할만한 동기부여가 있으면 좋겠다. ex) 시각화, 잔디밭, 댓글(소통)

# 미션
- 포도알 기능을 구현한다. 중심 키워드가 있고, 그 키워드를 통해 퍼져나가는 개념들을 이어서 정리한다. 다른 크루가 키워드(중심, 연결)를 검색하면, 포도알 전체를 보여준다! 
    -> 포도알 내용이 태그가 되면 어떨까요?
    -> 어떻게 시각화를 풀어가면 동기부여가 될수 있을까?
    -> 어떤 거를 공부했는지 동기부여도 되고, 꾸준함도 보상해주는 시각화도 잇으면 좋지 않을까! 
- 댓글 기능을 구현하여 소통의 장을 넓힌다!

**Comment by sunhpark42** [comment]
# 너잘, 서니, 티케

## 비전

크루들의 학습로그 작성에대한 부담을 줄여준다. 
or 크루들의 학습로그 작성에 대한 동기부여를 한다.

## 미션
* 학습로그 작성에 대한 부담을 줄여줄 수 있도록 정보를 나누고, 로드맵을 공유할 수 있는 서비스를 개발한다.
  * 양식의 자유화. => 자유도에 맡기자. 
  * 학습 로드맵을 제공한다.
  * 이전에는 과제를 통해서 정해진 길이 있는 거라고 생각함. (정해진 로드맵을 걷는 느낌) 레벨 3은 주차별로 내가 어디서 인사이트를 얻고, 어떤걸 공부했는지 알 수 있어서 적다보면, 정리된 목차만 보더라도 이 플젝을 통해서 뭘 배웠는지 한 눈에 볼 수 있길 바랬다. 
  * 토론장을 만들어서, 배포 공부해보고 싶은데, 뭐 공부 해볼까요? 이런것들? => 슬랙잡담에 올리는 것 처럼. (안사라진다는 장점 굿)
깃허브 디스커션느낌쓰.

* 학습로그 작성에 대한 동기부여가 되도록, 학습한 내용을 시각화한다.
   * 예시 ) 깃잔디찍기, 업적 시스템, 워드 맵 등
<img width="381" alt="스크린샷 2021-08-09 오후 5 31 13" src="https://user-images.githubusercontent.com/67677561/128678806-bfa5c49f-a46c-4793-a163-59baa698b4b1.png">
<img width="1027" alt="스크린샷 2021-08-09 오후 5 31 41" src="https://user-images.githubusercontent.com/67677561/128678871-466c0245-55da-49d7-9876-6111a8442316.png">
<img width="784" alt="스크린샷 2021-08-09 오후 5 32 19" src="https://user-images.githubusercontent.com/67677561/128678943-0ed8158a-1167-4121-8b00-7aa8b1834d7f.png">

**Comment by gracefulBrown** [comment]
## 비전
- 베타에서 벗어나 정식 서비스로 발돋움한다.
- TIL 대통합

## 미션
- 서비스 안정화
- 크루들의 학습로그 작성에 대한 부담을 줄여주고 작성에 대한 동기부여를 한다.

## 피쳐
- 토론 공간(DIscussion, 댓글)
- 내가 만드는 로드맵
- 포트폴리오
